tests/unit/acquisition/test_function.py

```diff
@@ -1692,18 +1738,29 @@ def prepare_acquisition_function(
         ) -> AcquisitionFunction:
             return lambda x: tf.ones_like(tf.squeeze(x, -2))
 
-    data = {"foo": Dataset(train_x, model_pred_observation)}
+    data = {"foo": Dataset(train_x[:5], model_pred_observation[:5])}
     models_ = {"foo": obj_model}
 
-    echvi = ExpectedConstrainedHypervolumeImprovement(
-        "foo", _Certainty(), 0
-    ).prepare_acquisition_function(data, models_)
+    builder = ExpectedConstrainedHypervolumeImprovement("foo", _Certainty(), 0)
+    echvi = builder.prepare_acquisition_function(data, models_)
 
     ehvi = ExpectedHypervolumeImprovement().using("foo").prepare_acquisition_function(data, models_)
 
     at = tf.constant([[[-0.1]], [[1.23]], [[-6.78]]])
     npt.assert_allclose(echvi(at), ehvi(at))
 
+    new_data = {"foo": Dataset(train_x, model_pred_observation)}
+    up_echvi = builder.update_acquisition_function(echvi, new_data, models_)
+    assert up_echvi == echvi
+    up_ehvi = (
+        ExpectedHypervolumeImprovement()
+        .using("foo")
+        .prepare_acquisition_function(new_data, models_)
+    )
```

Comment:
Why is that necessary, if basically the same happened two lines above?

Reply:
This is creating an updated ehvi function based on the new data to compare the updated echvi function (it's actually creating a brand new function but that's ok as we're not testing updating ehvi here, only echvi).

Reply:
ok, i see , thanks